python/paddle/fluid/layers/nn.py

```diff
@@ -14508,7 +14508,17 @@ def sign(x):
     helper = LayerHelper("sign", **locals())
 
     if not isinstance(x, Variable):
-        x = assign(x)
+        if isinstance(x, np.ndarray):
+            x = assign(x)
+        else:
+            raise TypeError(
+                "The type of 'x' in sign_op must be Variable or numpy.ndarray, but received %s."
+                % (type(x)))
+
+    if convert_dtype(x.dtype) not in ['float32', 'float64']:
+        raise TypeError(
+            "The data type of 'x' in sign_op must be float32, float64, but received %s."
```

Comment:
`float32, float64` -> `float32 or float64`

Reply:
OK, I will fix it.